### PR TITLE
remove cvmix, geokdtree and mom6_da_hooks from soca/bundle

### PR DESCRIPTION
**Pull request by aerorahul**
Change-Id: I1c50532d2e19a6871e53976632124f9f459ff21b

## Description

Provide a detailed description of what this PR does.  - This PR removes the cloning of CVMix, geoKdTree, and MOM6_da_hooks, and instead recursively clones these as part of MOM6.
What bug does it fix, or what feature does it add? - Removes overhead of maintaining 3rd party dependencies of MOM6 and allows us to be in near-constant sync with upstream development.
Is a change of answers expected from this PR? - Perhaps. We used to maintain separate forks of CVMix, geoKdTree and MOM6_da_hooks and never really updated the forks.  If they changed from the time we made those forks, answers are bound to change.

## Testing

How were these changes tested? - Waiting from Travis CI to tell me I made a mistake
What compilers / HPCs was it tested with? - Irrelevant
Are the changes covered by ctests? (If not, tests should be added first.) - Irrelevant

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names for the automated TravisCI tests to pass

Do PRs in upstream repositories need to be merged first? - Yes. Got rid of some dependencies, added those implicitly here.
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on JCSDA/mom6/pull/[14](https://github.com/JCSDA/MOM6/pull/14)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,14 +20,14 @@
 env:
   global:
     - MAIN_REPO=soca
-    - LIB_REPOS="fms cvmix geokdtree mom6_da_hooks gsw mom6 crtm
+    - LIB_REPOS="fms gsw crtm
                  fckit atlas oops saber ioda ufo ioda-converters soca-config"
     - BUILD_OPT=""
```
Comment (guillaumevernieres) [inline .travis.yml:25]: missing mom6

Comment (aerorahul) [reply .travis.yml:25]: really, I see it right there

**Comment by aerorahul** [comment]
@guillaumevernieres @travissluka 
I was able to successfully build `soca/bundle` on my `macOS`.

**Comment by guillaumevernieres** [comment]
> [...]
> ## Description
> 
> **Provide a detailed description of what this PR does. - No.**
> 
> useless blabla
>
> [...]

Outrageous !!!!

**Comment by guillaumevernieres** [comment]
Note to self and @aerorahul : That [script](https://github.com/JCSDA/soca/blob/develop/.github/travisci/build.sh) will need to be updated as well. I'll have a look at it, but we will still probably have to wait for a new container with the recursive cloning option in ecbuild. 
Thanks for doing that work!

Edit: Just checked, the recursive option is not in the container yet.
rec-not-in-container

@mmiesch , any idea when the container will be updated with a more recent version of ecbuild?

**Comment by mmiesch** [comment]
@guillaumevernieres - Containers exist now with new versions of ecbuild and eckit.  But, tests are currently failing so if we deploy them now they will break the CI.  Hopefully this will be solved by the end of the day today.  We are tentatively planning to deploy the new containers Monday morning at 10am MDT but this is contingent on getting all tests to pass.

**Comment by aerorahul** [comment]
Thanks @mmiesch. @guillaumevernieres I think this PR can really wait till all the travis-CI issues are ironed out.  The changes are only in the top-level `CMakeLists.txt` in SOCA and in the `src` directory of MOM6, and will be very easy to update at any point in the future.
However, changes to the `build.sh` script are going to be another issue.  I am not familiar with how the Travis-CI is set-up, so that may take time -- no more than a day I imagine.

**Review by guillaumevernieres — APPROVED** [review]
:tada: :tada:

**Pull request merged**